Lib/typing.py

```diff
@@ -1429,11 +1430,16 @@ def name_by_id(user_id: UserId) -> str:
         num = UserId(5) + 1     # type: int
     """
 
-    def new_type(x):
-        return x
+    NewTypeClass = type(name, (object,), {
+        '__supertype__' : tp,
+        '__name__' : name,
+        '__repr__' : lambda self: f"<{self.__name__}:{tp.__name__}>",
+        '__call__' : staticmethod(lambda x: x)
+    })
 
-    new_type.__name__ = name
-    new_type.__supertype__ = tp
+    new_type = NewTypeClass()
+    assert new_type.__name__ == name
+    assert new_type.__supertype__ is tp
```

Comment:
I'd argue that it's best you asserted this inside a test in `Lib/test/test_typing.py` instead of here since this adds checking overhead every time this is used?

Reply:
Sounds like a better deal, indeed. I’ll update the PR. Thanks!